src/modules/airdrop/services/airdrop-service.ts

```diff
@@ -123,6 +132,89 @@ export class AirdropService {
+
+    const recipientsFile = await readFile(file.path, { encoding: "utf8" });
+    const recipientsJson = JSON.parse(recipientsFile);
+
+    try {
+      return await this.dataSource.transaction(async (entityManager) => {
+        const window = await entityManager
+          .createQueryBuilder()
+          .insert()
+          .into(MerkleDistributorWindow)
+          .values({
+            chainId: recipientsJson["chainId"],
+            rewardToken: recipientsJson["rewardToken"],
+            windowIndex: recipientsJson["windowIndex"],
+            rewardsToDeposit: recipientsJson["rewardsToDeposit"],
+            merkleRoot: recipientsJson["merkleRoot"],
+          })
+          .execute();
+        const windowId = window.identifiers[0].id;
+
+        for (const recipient of Object.values(recipientsJson["recipientsWithProofs"])) {
+          await entityManager
+            .createQueryBuilder()
+            .insert()
+            .into(MerkleDistributorRecipient)
+            .values({
+              merkleDistributorWindowId: windowId,
+              address: recipient["account"],
+              amount: recipient["amount"],
+              accountIndex: recipient["accountIndex"],
+              proof: recipient["proof"],
+              payload: recipient["metadata"],
+            })
+            .execute();
+        }
+
+        const query = entityManager
+          .createQueryBuilder(MerkleDistributorRecipient, "recipient")
+          .select("COUNT(*) as count")
+          .where("recipient.merkleDistributorWindowId = :windowId", { windowId });
+        const recipientsCount = await query.getRawOne();
+
+        return {
+          recipients: isNaN(parseInt(recipientsCount.count)) ? null : parseInt(recipientsCount.count),
+        };
+      });
+    } catch (error) {
+      if (
+        error instanceof QueryFailedError &&
+        (error as QueryFailedError).driverError.constraint === "UK_merkle_distributor_window_windowIndex"
+      ) {
+        throw new DuplicatedMerkleDistributorWindowException();
+      }
+
+      throw error;
+    }
+  }
+
+  public async getMerkleDistributorProof(address: string, windowIndex: number) {
+    const checksumAddress = ethers.utils.getAddress(address);
+
+    const query = this.dataSource
+      .createQueryBuilder(MerkleDistributorRecipient, "recipient")
+      .innerJoinAndSelect("recipient.merkleDistributorWindow", "window")
+      .where("recipient.address = :address", { address: checksumAddress })
+      .andWhere("window.windowIndex = :windowIndex", { windowIndex });
+    const recipient = await query.getOne();
+
+    if (!recipient) return {};
+
+    return {
+      accountIndex: recipient.accountIndex,
+      address: recipient.address,
+      amount: recipient.amount,
+      payload: recipient.payload,
+      proof: recipient.proof,
+      merkleRoot: recipient.merkleDistributorWindow.merkleRoot,
+      windowIndex: recipient.merkleDistributorWindow.windowIndex,
+    };
```

Comment:
Can we also join the associated discord details (name, avatar) here from the user entity if it exists for the given address?

Reply:
Are they needed on the FE? Can you show me a screenshot?

Reply:
Yes sure. It is required when connecting an eligible wallet

<img width="627" alt="image" src="https://user-images.githubusercontent.com/17645687/196737142-120d0660-27f9-4093-bd75-cc34a95da873.png">

Reply:
I see. Ok, I'll add them. It is also the option to include this data in the metadata object part of the JSON file

Reply:
Yea either that or we can just add a separate endpoint that returns the connected discord account for a given address.

Reply:
Not sure if the discord details should be part of the JSON but could also make sense yea

Reply:
I will approve for now and this can be tackled separately